### ISSUE TITLE
podIdentity.provider : aws-kiam

### ISSUE THREAD

**Issue by elihai-risk** [kedacore/keda#645]
i set:

```yaml
Kind:         TriggerAuthentication
Metadata:
  Creation Timestamp:  2020-02-27T14:53:06Z
  Generation:          1
  Resource Version:    195722456
  Self Link:           /apis/keda.k8s.io/v1alpha1/namespaces/core/triggerauthentications/keda-trigger-auth-aws-credentials
  UID:                 xxxxxxxxxxxxxxx
Spec:
  Pod Identity:
    Provider:  aws-kiam
```

but still getting this error on the keda operator :
> {"level":"error","ts":1583070731.7554772,"logger":"controller-runtime.controller","msg":"Reconciler error",
"controller":"scaledobject-controller","request":"core/aws-sqs-queue-scaledobject-historical-queue-consumer
","error":"error getting scaler for trigger #0: Error parsing SQS queue metadata: 'AWS_ACCESS_KEY_ID' doesn
't exist in the deployment environment","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg
/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*C
ontroller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.2/pkg/internal/controller/co
ntroller.go:218\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n
\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.2/pkg/internal/controller/controller.go:192\nsigs.k8s.io/
controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/pkg/mod/sigs.k8s.io/controller-runti
me@v0.2.2/pkg/internal/controller/controller.go:171\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\
t/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:152\nk8s.io/apima
chinery/pkg/util/wait.JitterUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg
/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/pkg/mod/k8s.io/apimachinery@v0.0.0-2
0190404173353-6a84e37a896d/pkg/util/wait/wait.go:88"}

**Comment by tomkerkhove** [comment]
Can you provide your `ScaledObject` as well please?

**Comment by elihai-risk** [comment]
apiVersion: keda.k8s.io/v1alpha1
kind: ScaledObject
metadata:
  name: aws-sqs-queue-scaledobject 
  namespace: test 

    deploymentName: test
spec:
  scaleTargetRef:
    deploymentName: test
    pollingInterval: 30  # Optional. Default: 30 seconds
    cooldownPeriod:  300 # Optional. Default: 300 seconds
    minReplicaCount: 1   # Optional. Default: 0
    maxReplicaCount: 6 # Optional. Default: 100
  triggers:
  - type: aws-sqs-queue
    authenticationRef:
      name: keda-trigger-auth-aws-credentials
    metadata:
      queueURL: https://sqs.us-east-1.amazonaws.com/xxxxxxxxx/xxxxxxxx
      queueLength: "1000"
      awsRegion: "eu-west-1"

**Comment by tomkerkhove** [comment]
It looks like the doc PR got merged already but feature PR is still open on https://github.com/kedacore/keda/pull/499

So that means it's not shipped yet.

**Comment by tomkerkhove** [comment]
Support is available as of v1.3.0
